### PR TITLE
Include yaml examples in dataset docs for common Datasets 

### PR DESCRIPTION
**Pull request by avan-sh**
## Description
Adds yaml examples in dataset docs as discussed in #920 & #579. 

## Development notes
Added yaml docs for the following datasets. I've put up the rendered docs in a temporary branch which can be viewed [here](https://htmlpreview.github.io/?https://raw.githubusercontent.com/avan-sh/kedro/temp/render_docs/docs/build/html/kedro.extras.datasets.pandas.CSVDataSet.html).

Pandas Datasets:
 - kedro.extras.datasets.pandas.CSVDataSet
 - kedro.extras.datasets.pandas.ExcelDataSet
 - kedro.extras.datasets.pandas.AppendableExcelDataSet
 - kedro.extras.datasets.pandas.GBQTableDataSet
 - kedro.extras.datasets.pandas.HDFDataSet
 - kedro.extras.datasets.pandas.JSONDataSet
 - kedro.extras.datasets.pandas.ParquetDataSet
 - kedro.extras.datasets.pandas.SQLQueryDataSet : **Added an extra entry for credentials config.** [docs](https://htmlpreview.github.io/?https://raw.githubusercontent.com/avan-sh/kedro/temp/render_docs/docs/build/html/kedro.extras.datasets.pandas.SQLQueryDataSet.html)
 - kedro.extras.datasets.pandas.SQLTableDataSet: **Added an extra entry for credentials config.** [docs](https://htmlpreview.github.io/?https://raw.githubusercontent.com/avan-sh/kedro/temp/render_docs/docs/build/html/kedro.extras.datasets.pandas.SQLTableDataSet.html)

Spark Datasets
 - kedro.extras.datasets.spark.SparkDataSet
 - kedro.extras.datasets.spark.SparkJDBCDataSet
 - kedro.extras.datasets.spark.SparkHiveDataSet

Other:
 - kedro.extras.datasets.pickle.PickleDataSet 
 - kedro.io.PartitionedDataSet: **Added an extra hyperlink for partitioned dataset**. Docs [here](https://raw.githubusercontent.com/avan-sh/kedro/temp/render_docs/docs/build/html/kedro.io.PartitionedDataSet.html)

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by datajoely** [comment]
This is looking great! Can you send a screenshot of what it looks like rendered!

File: kedro/extras/datasets/pandas/csv_dataset.py [hunk 1]
```diff
@@ -49,7 +49,30 @@ class CSVDataSet(AbstractVersionedDataSet):
     """``CSVDataSet`` loads/saves data from/to a CSV file using an underlying
     filesystem (e.g.: local, S3, GCS). It uses pandas to handle the CSV file.
 
-    Example:
+    Example adding a catalog entry with
+    `YAML API <https://kedro.readthedocs.io/en/stable/05_data/01_data_catalog.html#using-the-data-catalog-with-the-yaml-api>`_:
```
[the remark this answers is not in the document]
Comment (avan-sh) [reply kedro/extras/datasets/pandas/csv_dataset.py:53]: I missed the linters failing part, I'll how I can break it into multiple lines.

**Comment by WaylonWalker** [comment]
👏👏 Yes, thanks for the great work!

**Review by AntonyMilneQB — APPROVED** [review]
Sorry it's taken so long to get round to reviewing this but this is amazing work, thank you so much for the contribution! This is really a huge improvement.

I've left some minor improvements (mainly to simplify/tidy it and remove comments where they're not so relevant to the particular dataset). Please feel free to merge once you've made the changes. If the builds are failing then let me know and I'll take a look since we definitely want to get this merged 🙂

**Comment by WaylonWalker** [comment]
I was coaching a new Data Scientist on my team today, and noticed that the docs have yaml and I was so excited!!!

**Pull request merged**